cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/ProgramRuntimeProviderLoader.java

```diff
@@ -87,7 +68,7 @@ public ProgramRuntimeProviderLoader(CConfiguration cConf) {
   @Nullable
   public ProgramRuntimeProvider get(ProgramType programType) {
     try {
-      ProgramRuntimeProvider provider = programRunnerProviderCache.get(programType);
+      ProgramRuntimeProvider provider = programRuntimeExtensionLoader.getExtension(programType);
       if (provider != NOT_SUPPORTED_PROVIDER) {
```

Comment:
Yes, the existing contract of the ``ProgramRuntimeProviderLoader`` was that it first tries to load the extension from the extension classloader. If it is not found, it loads from the system classloader. If it is not found there, it returns ``NOT_SUPPORTED_PROVIDER``